### PR TITLE
build: upgrade thirdparty to 0.6.0

### PR DESCRIPTION
**Pull request by aceforeverd**
1. upgrade thirdparty, which upgrade brpc to 1.6.0
2. calls some teardown function before nameserver destructor, it mitigate cases like #3416

### CODE REVIEW

File: src/nameserver/name_server_impl.cc [hunk 1]
```diff
@@ -5004,19 +5006,21 @@ void NameServerImpl::UpdateTableStatus() {
...
-        if (!kv.second->client_->GetTableStatus(tablet_status_response)) {
-            PDLOG(WARNING, "get table status failed! endpoint[%s]", kv.first.c_str());
-            continue;
-        }
-        for (int pos = 0; pos < tablet_status_response.all_table_status_size(); pos++) {
-            std::string key = absl::StrCat(tablet_status_response.all_table_status(pos).tid(), "_",
-                              tablet_status_response.all_table_status(pos).pid(), "_", kv.first);
-            pos_response.emplace(key, tablet_status_response.all_table_status(pos));
+
+        for (const auto& kv : tablet_ptr_map) {
```
Comment (dl239) [inline src/nameserver/name_server_impl.cc]: what's the reason for the update?

**Review by tobegit3hub — APPROVED** [review]
LGTM

File: src/nameserver/name_server_impl.cc (hunk 1 above)
Comment (aceforeverd) [reply src/nameserver/name_server_impl.cc]: see #3416. I just forward the workaround previously, without it `single_tablet_test` will not teardown correctly. 
PLS check if there are any side effects

Comment (dl239) [reply src/nameserver/name_server_impl.cc]: there are some problems if send rpc request under the lock

Comment (aceforeverd) [reply src/nameserver/name_server_impl.cc]: reverted

**Review by dl239 — APPROVED** [review]

**Pull request merged**